aws-lc-rs/src/buffer.rs

```diff
@@ -0,0 +1,85 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR ISC
+
+//! This module exposes a buffer type used in crate APIs returning private keys and other "private"
+//! contents.
+
+#![allow(clippy::module_name_repetitions)]
+
+use std::borrow::Cow;
+use std::fmt;
+use std::marker::PhantomData;
+
+use zeroize::Zeroize;
+
+/// This is a buffer type for some data exposed by various APIs in this crate.
+///
+/// `T` acts as a discriminant between different kinds of data.
+///
+/// The buffer will be zeroed on drop if it is owned.
+pub struct Buffer<'a, T>(Cow<'a, [u8]>, PhantomData<T>);
```

Comment:
Should we maybe call this `SecretBuffer` or `PrivateBuffer` I could see this type being useful for other data like public-key data etc. But the description implies that the content is meant to be private. Should this be communicated through the name, or should the documentation be updated to reflect that the data MAY be private.

Reply:
>But the description implies that the content is meant to be private.

Note FIPS140-3 Security Level 4 requires zeroisation of all unprotected SSPs (Security Sensitive Parameters), where SSP includes PSP (Public Security Parameters).  In other words, the need to zeroize a buffer do NOT necessarily mean the buffer contains any secret/private data.  It may also contain public data, such as public key, and still may need to be zeroised depending on the specific context.

In short, "the description implies that the content is meant to be private" is not technically correct.